Session 1

## **Mathematical Manipulations**

*   Addition
*   Subtraction
*   Multiplication
*   Division







***Variable declaration***

In [ ]:
a=10
b=8

adding the two numbers

In [ ]:
a+b

18

subtracting the two numbers

In [ ]:
a-b

2

multiplication of the two numbers

In [ ]:
a*b

80

division of the two numbers

In [ ]:
a/b

1.25

***Improved result display***

In [ ]:
print('The sum of',a, 'plus', b, 'is',a+b)

The sum of 10 plus 8 is 18


A being 10 and b being 8 their sum will add upto 18

In [ ]:
from re import T
print('the difference between',a,'and',b,'is',a-b)

the difference between 10 and 8 is 2


A being 10 and b being 8 their difference is 2

In [ ]:
print('The product between',a,'and',b,'is',a*b)

The product between 10 and 8 is 80


A being 10 and b being 8 their product is 80

In [ ]:
print('The quotient of ',a,'and',b,'is',a/b)

The quotient of  10 and 8 is 1.25


A being 10 and b being 8 their quotient is 1.25

division with modulus remainder

In [ ]:
c=17
d=3

In [ ]:
c/d

5.666666666666667

In [ ]:
c//d

5

In [ ]:
c%d

2

In [ ]:
print('the quotient of',c,'and',b, 'is',c/b)

the quotient of 17 and 3 is 5.666666666666667


In [ ]:
print('the remainder after dividing',c,'and',d,'is',c%d)

the remainder after dividing 17 and 3 is 2


In [ ]:
print('The quotient of',c, 'and',d,'is',c//d,'remainder',c%d)

The quotient of 17 and 3 is 5 remainder 2


***Exponents***

In [ ]:
base=8
exponent=3

In [ ]:
base**exponent

512

In [ ]:
print('8 raised to power 3 is',base**exponent )

8 raised to power 3 is 512


***Area of a rectangle***

In [ ]:
width=5
height=4

In [ ]:
width*height

20

In [ ]:
print('The area of a rectangle width 5 and height 4 is',width*height)

The area of a rectangle width 5 and height 4 is 20


***REVISION***

***Area of a rectangle***

In [ ]:
breath=6
extent=4

In [ ]:
breath*extent

24

In [ ]:
print("The area of a rectangle breath 6 and extent 4 is",breath*extent)

The area of a rectangle breath 6 and extent 4 is 24


***Exponents***

In [ ]:
base=6
exponent=2

In [ ]:
base**exponent

36

In [ ]:
print("6 raised to power 2 is",base**exponent)

6 raised to power 2 is 36


6 raised to power 2 is 36

Division with modulus remainder

In [ ]:
e=5
f=3

In [ ]:
e/f

1.6666666666666667

In [ ]:
e//f

1

In [ ]:
e%f

2

In [ ]:
print("The quotient of ",e, "and",f,"is",e/f)

The quotient of  5 and 3 is 1.6666666666666667


In [ ]:
print('The remainder after dividing',e,"and",f, "is",e%f)

The remainder after dividing 5 and 3 is 2


In [ ]:
print("The quotient of",e,"and",f,"is",e//f, "remainder",e%f)

The quotient of 5 and 3 is 1 remainder 2


***Area of a triangle***

In [ ]:
base=4
height=2

In [ ]:
1/2*base*height

4.0

In [ ]:
print('The area of a triangle with base 4 and height 2 is',1/2*base*height)

The area of a triangle with base 4 and height 2 is 4.0


***Simple interest***

In [ ]:
rate=.1
principal=50000 # in dollars
time=2 # in years

In [ ]:
interest=principal*rate *time

In [ ]:
interest

10000.0

In [ ]:
print('The simple interest for an amount of 50000 dollars at a rate of 10% over a period of 2 years is',interest )

The simple interest for an amount of 50000 dollars at a rate of 10% over a period of 2 years is 10000.0


***Temperature conversion(celsius to fahrenheit)***

In [ ]:
celsius=23

In [ ]:
temp=(celsius *9/5) +32

In [ ]:
print('The conversion of celsius 23 to fahrenheit',temp)

The conversion of celsius 23 to fahrenheit 73.4


***Average of numbers***

In [ ]:
num1=3
num2=6
num3=9

In [ ]:
average=(num1 +num2+num3)/3
average

6.0

In [ ]:
(num1 +num2+num3)/3

6.0

In [ ]:
print('The average of num1 and num2 and num3 is',average)

The average of num1 and num2 and num3 is 6.0


***Swaping numbers***

In [ ]:
a=4
b=6

In [ ]:
a,b=b,a

In [ ]:
a

6

Write a program that returns the largest number from a pool of 3

In [ ]:
# Step 1: Input three numbers
a = float(input("Enter first number: "))
b = float(input("Enter second number: "))
c = float(input("Enter third number: "))

# Step 2: Compare the numbers
if a >= b and a >= c:
    print("The largest number is", a)
elif b >= a and b >= c:
    print("The largest number is", b)
else:
    print("The largest number is", c)

Enter first number: 5
Enter second number: 7
Enter third number: 2
The largest number is 7.0


Session 2

**elif else conditions**

In [ ]:
score=70

In [ ]:
if score >=90:
   print('A')
elif score >=80:
   print('B')
elif score >=70:
   print('C')
elif score >=60:
   print('D')
else:
   print('fail')

C


In [ ]:
score=90

In [ ]:
if score >=90:
   print('A')
elif score >=80:
   print('B')
elif score >=70:
   print('C')
elif score >=60:
   print('D')
else:
   print('Fail')

A


Loops:for and while

In [ ]:
for i in range(1,8):
  print('Hello',i)

Hello 1
Hello 2
Hello 3
Hello 4
Hello 5
Hello 6
Hello 7


Functions

greetings by definition

In [ ]:
def greet(name): #defining the greet function
  return f'Hello, {name}!'# calling ang applying the greet function
print(greet('Alice')) #display the results

Hello, Alice!


In [ ]:
def greet(world):
  return f'Hello, {world}!'
print(greet('world'))

Hello, world!


In [ ]:
fruits=['apple','banana','mango']#to list the items
fruits.append('kiwi') #to add an item to the list
fruits.sort() #to sort the items in an alphabetical order

In [ ]:
fruits

['apple', 'banana', 'kiwi', 'mango']

***LARGEST POOL OF 3***

In [ ]:
a=4
b=7
c=1

In [ ]:
if a>=b and a>=c:
   largest=('a')
if b>=a and b>=c:
  largest=('b')
else:
 largest=('c')

In [ ]:
largest

'b'

***elif if condition***

In [ ]:
purchase =200 #in ksh

In [ ]:
if purchase>=500:
  print("30% discount")
  discount=('30%')
else:
  print('0% discount')

0% discount


In [ ]:
discount=purchase*discount

In [ ]:
weather=("g")

In [ ]:
if weather is ("sunny"):
   print("Its really sunny today")
else:
  print("Its not sunny today")

# You are supposed to declare/define weather

Its not sunny today


<>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
<>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
<ipython-input-35-d5f4e703b622>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if weather is ("sunny"):


In [ ]:
age=13

In [ ]:
if age>=18:
  print("An adult")
else:
  print("Not an adult")

Not an adult


***session three***

Creating comprehensive listn (list comprehension)

In [ ]:
# Let i,j be a sub space of intergers where both ranges from 1 to 4,
#create a table of multiplyers i by j
squares=[]
for i in range(1,4):
   for j in range(1,4):
    print(f"{i}x{j}={i*j}")

1x1=1
1x2=2
1x3=3
2x1=2
2x2=4
2x3=6
3x1=3
3x2=6
3x3=9


In [ ]:
#traditional way
squares=[]
for i in range(1,6):
  squares.append(i*i)
  print(squares)

[1]
[1, 4]
[1, 4, 9]
[1, 4, 9, 16]
[1, 4, 9, 16, 25]


**Assignment**

Multiplication table for i,j where i and j is a set of numbers ranging from 1 to 15

In [ ]:
squares=[]
for i in range(1,15):
  for j in range(1,15):
    print(f"{i}x{j}={i*j}")

1x1=1
1x2=2
1x3=3
1x4=4
1x5=5
1x6=6
1x7=7
1x8=8
1x9=9
1x10=10
1x11=11
1x12=12
1x13=13
1x14=14
2x1=2
2x2=4
2x3=6
2x4=8
2x5=10
2x6=12
2x7=14
2x8=16
2x9=18
2x10=20
2x11=22
2x12=24
2x13=26
2x14=28
3x1=3
3x2=6
3x3=9
3x4=12
3x5=15
3x6=18
3x7=21
3x8=24
3x9=27
3x10=30
3x11=33
3x12=36
3x13=39
3x14=42
4x1=4
4x2=8
4x3=12
4x4=16
4x5=20
4x6=24
4x7=28
4x8=32
4x9=36
4x10=40
4x11=44
4x12=48
4x13=52
4x14=56
5x1=5
5x2=10
5x3=15
5x4=20
5x5=25
5x6=30
5x7=35
5x8=40
5x9=45
5x10=50
5x11=55
5x12=60
5x13=65
5x14=70
6x1=6
6x2=12
6x3=18
6x4=24
6x5=30
6x6=36
6x7=42
6x8=48
6x9=54
6x10=60
6x11=66
6x12=72
6x13=78
6x14=84
7x1=7
7x2=14
7x3=21
7x4=28
7x5=35
7x6=42
7x7=49
7x8=56
7x9=63
7x10=70
7x11=77
7x12=84
7x13=91
7x14=98
8x1=8
8x2=16
8x3=24
8x4=32
8x5=40
8x6=48
8x7=56
8x8=64
8x9=72
8x10=80
8x11=88
8x12=96
8x13=104
8x14=112
9x1=9
9x2=18
9x3=27
9x4=36
9x5=45
9x6=54
9x7=63
9x8=72
9x9=81
9x10=90
9x11=99
9x12=108
9x13=117
9x14=126
10x1=10
10x2=20
10x3=30
10x4=40
10x5=50
10x6=60
10x7=70
10x8=80
10x9=90
10x10=100
10x11=110

**Basic dictionary **

Used to pull information from a data base that satisfies a given condition


FUNCTIONS WITH CONDITIONS

In [ ]:
#a program that checks the eligibility of a voter
def is_eligible(age):
  if age>=18:
    return "Eligible to vote"
  else:
    return "Not eligible to vote"
print(is_eligible(14))

Not eligible to vote


Assignment

In another field


In [ ]:
def risk_category(age,cholesterol,smoker):
  if age>60 and cholesterol>240 and smoker:
    return "High risk"
  elif age>45 and cholesterol>200:
    return "Moderate risk"
  else:
    return "Low risk" # You need to add a print function after every program to check if it works as desired. Refer to the code below.

In [ ]:
print(risk_category(30, 250, False))

Low risk


**Basic dictionary operations**

In [ ]:
student = {"name": "Alice", "age": 20, "grade": "B"} # Supplying the demographic information of a student

print(student["name"])# Telling the python to print the name of the student
student["grade"] = "A"# Indicating the grade a specific student attained
print(student)# Printing out every information about the student

# Add new key
student["school"] = "St. Mary" # Adding  anew identifier to the list of an individual

Alice
{'name': 'Alice', 'age': 20, 'grade': 'A'}


In [ ]:
for key, value in student.items(): # Used to display independent details/information of an individual in an independent format.
  print(key,"is",value)

name is Alice
age is 20
grade is A
school is St. Mary


Assignment

basic dictionary operations in another field

In [ ]:
patient={"name":"John","age":32,"medication":"injection"}

print(patient["name"])
patient["medication"]="injectons"
print(patient)

John
{'name': 'John', 'age': 32, 'medication': 'injectons'}


Read on creating a menu program using loop+input

***Password generator***

In [ ]:
import string
import random
import numpy as np
import torch


def generate_password(length=8,use_upper=True,use_lower=True,use_symbols=True,use_digits=True):
  character=''
  if use_upper:
    character+=string.ascii_uppercase
  if use_lower:
    character+=string.ascii_lowercase
  if use_symbols:
    character+=string.punctuation
  if use_digits:
    character+=string.digits

  if not character:
     raise ValueError("At least one character set must be selected!")

  #Ensure password contains at least one of each selected type
  password=[]
  if use_upper:
    password.append(random.choice(string.ascii_uppercase))
  if use_lower:
    password.append(random.choice(string.ascii_lowercase))
  if use_symbols:
    password.append(random.choice(string.punctuation))
  if use_digits:
    password.append(random.choice(string.digits))

  #Fill the rest of the password length
  password+=random.choices(character,k=length - len(password))
  random.shuffle(password)

  return ''.join(password)

In [ ]:
random.seed(32) #is used for reproducibility
print("generated password:",generate_password(length=8))

generated password: DCg4C*.W


***Making a calculator***

In [ ]:
def add(x, y):
    return x + y

def subtract(x, y):
    return x - y

def multiply(x, y):
    return x * y

def divide(x, y):
    if y == 0:
        return "Error! Cannot divide by zero!"
    return x / y

def calculator():
    print("Simple Calculator")
    print("Select operation:")
    print("1. Add")
    print("2. Subtract")
    print("3. Multiply")
    print("4. Divide")

    choice = input("Enter choice (1/2/3/4): ")

    if choice not in ['1', '2', '3', '4']:
        print("Invalid input.")
        return

    try:
        num1 = float(input("Enter first number: "))
        num2 = float(input("Enter second number: "))
    except ValueError:
        print("Invalid number input.")
        return

    if choice == '1':
        print(f"Result: {add(num1, num2)}")
    elif choice == '2':
        print( f"Result: {subtract(num1, num2)}")


    elif choice == '3':
        print(f"Result: {multiply(num1, num2)}")
    elif choice == '4':
        print(f"Result: {divide(num1, num2)}")

if __name__ == "__main__":
    calculator()


Simple Calculator
Select operation:
1. Add
2. Subtract
3. Multiply
4. Divide
Enter choice (1/2/3/4): 4
Enter first number: 4
Enter second number: 7
Result: 0.5714285714285714


In [ ]:
import math

def trig_calculator():
  print("Trigonometry calculator")
  print("Available functions: sin, cos , tan , cosec , sec ,cot")

  angle_deg = float(input("Enter the angles in degrees:"))
  function = input("Enter the function name(e.g., sin, cos):").strip().lower()

  angle_rad = math.radians(angle_deg)

  try:
    if function == "sin":
      result = math.sin(angle_rad)
    elif function == "cos":
      result = math.cos(angle_rad)
    elif function == "tan":
      result = math.tan(angle_rad)
    elif function == "cosec":
      # cosec(x) = 1/sin(x)
      result = 1 / math.sin(angle_rad)
    elif function == "sec":
      # sec(x) = 1/cos(x)
      result = 1 / math.cos(angle_rad)
    elif function == "cot":
      # cot(x) = 1/tan(x) or cos(x)/sin(x)
      result = 1 / math.tan(angle_rad)

    else:
      print("Invalid function.")
      return

    print(f"{function}({angle_deg}) = {result:.6f}")
  except ZeroDivisionError:
    # Handle cases where the denominator is zero for cosec, sec, or cot
    print(f"{function}({angle_deg}) is undefined (division by zero).")

trig_calculator()

In [ ]:
import math

def trig_calculator():
  print("Trigonometry calculator")
  print("Available functions: sin, cos , tan , cosec , sec ,cot")

  angle_deg=float(input("Enter the angles in degress:"))
  function=input("Enter the function name(e.g., sin, cos):").strip().lower()

  angle_rad=math.radians(angle_deg)

  try:
    if function=="sin":
      result=math.sin(angle_rad)rt math
    elif function=="cos":
      result=math.cos(angle_rad)
    elif function=="tan":
      result=math.tan(angle_rad)
    elif function=="cosec":
      result=1 / math.sin(angle_rad)
    elif function=="sec":
      result=1 / math.cos(angle_rad)
    elif function=="cot":
      result=1 / math.tan(angle_deg)

    else:
      print("Invalid function.")
      return

    print(f"{function}({angle_deg})={result:.6f}")
  except ZeroDivisionError:
    print(f"{function}({angle_deg}) is undefined(divison by zero).")

trig_calculator()

Trigonometry calculator
Available functions: sin, cos , tan , cosec , sec ,cot
Enter the angles in degress:90
Enter the function name(e.g., sin, cos):cos
cos(90.0)=0.000000


combine the 2 programs and work on the user interface

In [ ]:
import math

def evaluate_trig_function(expr):
    # Expected format: func(angle), e.g. sin(30), cot(45)
    try:
        func_name, rest = expr.split('(', 1)
        angle_str = rest.rstrip(')')
        angle_deg = float(angle_str)
        angle_rad = math.radians(angle_deg)

        if func_name == "sin":
            return math.sin(angle_rad)
        elif func_name == "cos":
            return math.cos(angle_rad)
        elif func_name == "tan":
            return math.tan(angle_rad)
        elif func_name == "cosec":
            return 1 / math.sin(angle_rad)
        elif func_name == "sec":
            return 1 / math.cos(angle_rad)
        elif func_name == "cot":
            return 1 / math.tan(angle_rad)
        else:
            print("Invalid trig function.")
            return None
    except Exception as e:
        print("Error in trig expression:", e)
        return None

def is_trig_expression(expr):
    # Check if expr matches trig pattern like sin(30)
    trig_funcs = ['sin', 'cos', 'tan', 'cosec', 'sec', 'cot']
    for func in trig_funcs:
        if expr.startswith(func + '(') and expr.endswith(')'):
            return True
    return False

def calculator():
    print("Simple Calculator — enter expressions like '2+3*4' or trig functions like 'sin(30)'.")
    print("Type 'clear' to reset, 'exit' to quit.")

    expression = ""

    while True:
        print("\nScreen: " + expression)
        user_input = input("Enter next input (number/operator/trig function/ = / clear / exit): ").strip()

        if user_input.lower() == 'exit':
            print("Exiting calculator. Bye!")
            break
        elif user_input.lower() == 'clear':
            expression = ""
            continue
        elif user_input == '=':
            if is_trig_expression(expression):
                result = evaluate_trig_function(expression)
                if result is not None:
                    print(f"Result: {result:.6f}")
                expression = ""
            else:
                try:
                    # Evaluate normal arithmetic
                    result = eval(expression)
                    print(f"Result: {result}")
                except Exception as e:
                    print("Error evaluating expression:", e)
                expression = ""
        else:
            # Add input to expression
            expression += user_input

if __name__ == "__main__":
    calculator()


Simple Calculator — enter expressions like '2+3*4' or trig functions like 'sin(30)'.
Type 'clear' to reset, 'exit' to quit.

Screen: 


In [ ]:
# Program make a simple calculator that can add, subtract, multiply and divide using functions

# define functions
def add(x, y):
   """This function adds two numbers"""

   return x + y

def subtract(x, y):
   """This function subtracts two numbers"""

   return x - y

def multiply(x, y):
   """This function multiplies two numbers"""

   return x * y

def divide(x, y):
   """This function divides two numbers"""

   return x / y

# take input from the user
print("Select operation.")
print("1.Add")
print("2.Subtract")
print("3.Multiply")
print("4.Divide")

choice = input("Enter choice(1/2/3/4):")

num1 = int(input("Enter first number: "))
num2 = int(input("Enter second number: "))

if choice == '1':
   print(num1,"+",num2,"=", add(num1,num2))

elif choice == '2':
   print(num1,"-",num2,"=", subtract(num1,num2))

elif choice == '3':
   print(num1,"*",num2,"=", multiply(num1,num2))

elif choice == '4':
   print(num1,"/",num2,"=", divide(num1,num2))
else:
   print("Invalid input")